### DISCUSSION TITLE
Ghidra DEV setup for running within Eclipse

### DISCUSSION THREAD

**Discussion by Wall-AF** [NationalSecurityAgency/ghidra#3837] [Q&A]
What's the minimum `gradle ...` commands necessary to be able to run Ghidra from within Eclipse after a `git clone ...` or recent `git pull`?

**Comment by ryanmkurtz** [comment]
After a clone:
```
$ gradle -I gradle/support/fetchDependencies.gradle prepdev eclipse buildNatives
```

After a pull (needed based on what's changed, but safe to do always):
```
$ gradle prepdev cleanEclipse eclipse buildNatives
```

**Reply by astrelsky** [reply]
How do you feel about merging those tasks for after a pull? It would pretty much just be a do nothing task depending on those. Just so it is that much easier to setup a dev environment with eclipse even though eclipse is trash.

Can call it "refreshDev". A refreshed dev is a good dev ;)

**Reply by ryanmkurtz** [reply]
What would your full command look like?  Were you hoping to incorporate `fetchDependencies` too?

**Reply by astrelsky** [reply]
> What would your full command look like?  Were you hoping to incorporate `fetchDependencies` too?

Hmm, the `fetchDependencies` is an init script though. I'm not sure if you can include it. You could probably check them and stop with a message that it needs to be run if out of date.
